sbthost/nsc/src/main/scala/scala/meta/internal/sbthost/RangeOps.scala

```diff
@@ -0,0 +1,35 @@
+package scala.meta.internal.sbthost
+
+import scala.reflect.internal.{util => gu}
+import scala.meta.internal.{semanticdb3 => s}
+
+trait RangeOps { self: DatabaseOps =>
+  implicit class XtensionGPositionSRange(gpos: gu.Position) {
+    private def source: gu.BatchSourceFile = {
+      gpos.source.asInstanceOf[gu.BatchSourceFile]
+    }
+
+    private def line(offset: Int): Int = {
+      // NOTE: Looks like some seriously crazy stuff is going on here.
+      // source.offsetToLine(offset)
+      0
```

Comment:
I expanded on "crazy stuff" and included a comment explaining why this hack works for scala.meta.Position but is incompatible with how LSP treats line/character pairs.

Reply:
In LSP, characters=Int.MaxValue falls back to the last character in the line https://microsoft.github.io/language-server-protocol/specification#position